### DISCUSSION TITLE
Using LastPartitionMapping without an asset

### DISCUSSION THREAD

**Discussion by cleboo** [dagster-io/dagster#13918] [Q&A]
Hi!

I've been trying to find out how to select the last partition of an asset inside a graph. The background is this: I want to load the contents of the last partition of an asset into an external system without having to materialize that last partition once again.

To illustrate, here is an example of what I want to do using an extra asset:
```python
from dagster import (
    asset,
    op,
    graph,
    DailyPartitionsDefinition,
    OpExecutionContext,
    In,
    AssetIn,
    LastPartitionMapping,
    Definitions
)

partition = DailyPartitionsDefinition(start_date="2023-04-19")

@asset(partitions_def=partition)
def val(context: OpExecutionContext) -> str:
    return context.asset_partition_key_for_output()

@asset(ins={'val': AssetIn(partition_mapping=LastPartitionMapping())})
def val_last(context: OpExecutionContext, val: str):
    return val

@op(ins={'s': In()})
def add_string(context, s):
    """add --added to the string and log it"""
    context.log.info(str(s))
    s = s + "--added"
    context.log.info(s)
    return s

@graph
def generate_date_string():
    return add_string(
        val_last.to_source_asset()
    )


job = generate_date_string.to_job(
    "generate_last_date_string"
)

defs = Definitions(
    jobs=[job, ],
    assets=[val, val_last]
)
```
So `val_last` selects the last of `val`'s and that is used in the `generate_val_string` Job. Is there a way to avoid the `val_last` asset and instead select the last partition in the graph somehow. I would imagine something like this:
```python
from dagster import (
    asset,
    op,
    graph,
    DailyPartitionsDefinition,
    OpExecutionContext,
    In,
    AssetIn,
    LastPartitionMapping,
    Definitions
)

partition = DailyPartitionsDefinition(start_date="2023-04-19")

@asset(partitions_def=partition)
def val(context: OpExecutionContext) -> str:
    return context.asset_partition_key_for_output()

@op(ins={'s': In()})
def add_string(context, s):
    """add --added to the string and log it"""
    context.log.info(str(s))
    s = s + "--added"
    context.log.info(s)
    return s

@graph
def generate_val_string():
    return add_string(
        val.to_source_asset(partition_mapping=LastPartitionMapping())
    )


job = generate_date_string.to_job(
    "generate_last_date_string"
)

defs = Definitions(
    jobs=[job, ],
    assets=[val, ]
)
```
Or maybe the `to_job` function is the right place for this ... 

I noticed, I can access all the partitions by simply accepting a dictionary in the `add_string` op, but that requires me to load all the partitions when I'm only really interested in the last one. 

Can somebody point me in the right direction? 🙂 Or is this not possible (yet)?

**Comment by sryza** [comment]
Hey @cleboo - this looks relevant to https://github.com/dagster-io/dagster/issues/13357. Currently, there's not a way to specify a PartitionMapping when a job depends on a partitioned asset. If you need a workaround, the best thing you can do currently is probably use `load_asset_value` to directly load the asset from inside the `add_string` op: https://docs.dagster.io/concepts/assets/software-defined-assets#loading-asset-values-outside-of-dagster-runs.

**Reply by cleboo** [reply]
Hi @sryza !

> this looks relevant to https://github.com/dagster-io/dagster/issues/13357

Yes, I actually ran into the same issue so it is definitely related. I think it is a separate feature though, so I filed a feature request here: https://github.com/dagster-io/dagster/issues/13940

In the meantime, thanks for the workaround - I'll try that out - and thanks a lot for your help!
